src/main/webapp/dashboard.html

```diff
@@ -21,12 +21,55 @@
             });
         function appendData(data) {
```

Comment:
We don't need this function. It is just repeating what OpportunityGetServlet is doing. I just saw your comment on that PR. The idea was OpportunityGetServlet would return opporuntiy in proper html format instead of json, so that we can just print whatever that servlets returns inside a div or something on the frontend side. Please remove any dashboard related code in that servlet that does not belong there and use that servlet. Just send that servlet a get request with id of the opportunity and it will return the opportunity in html formating. Also use opportunity css file in css folder for css styling.

Reply:
Also opportunityGetServlet is handling things like otherRequirements list, which you are missing here.

Reply:
> Also opportunityGetServlet is handling things like otherRequirements list, which you are missing here.

Yeah, I am aware. I am adding sections one-by-one

Reply:
> We don't need this function. It is just repeating what OpportunityGetServlet is doing. I just saw your comment on that PR. The idea was OpportunityGetServlet would return dynamically generated html page instead of json, so that we can just print whatever that servlets returns inside a div or something on the frontend side. Please remove any dashboard related code in that servlet that does not belong there and use that servlet. Just send that servlet a get request with id of the opportunity and it will return the opportunity in html formating. You can change that servlet to modify html to however you like. Also use opportunity css file in css folder for css styling.

We actually don't need formatted output from back-end. Already made html pages are hard to alter / render. Since I am working on version two, instead of editing your files, I am writing separate files with different back-end logic. As a best practice, we receive objects from back-end and render them on front-end. As far as get all opportunities is concerned, I will try make it more efficient by calling opportunities on-demand with ids.

Reply:
Let's discuss some pros and cons of eafh approch. Here is why I rendering it on backend.
1-Instead of going from database->json->html, we can just do database->html
2-The opportunity has some complex data types like array list (which we need to split) and Date, java is better at handling those.
3-We can modify to java servlet to modify how html page, I don't understand what do you mean Its hard to edit.
Can you please tell me why as a best practice we receive object and render it on front end ? (Full stack is not my strong side) As far as I know there is client side rendering but there is also server side rendering and both are best practices.

Reply:
It's usually preferable to build an api style server which returns json (basically just the requested data without styling), this way you can decouple your design and formatting from core data handling logic. This way you can build different clients that simply use the api for data retrieval. In our case, if in future we decided to build an app version, we could do so without doing any changes on server.
Even if we don't build an app, an expert in design or frontend will be able to work on rendering independently and would not need to know the backend logic. I do see the point of having server side rendering some time but even in that case you'd decompose the software internally such that the rendering module consumes the data module.

for your points here are my thoughts:
1. Yeah, it's possible to do database->html but going through json provides you more power and makes the system easy to reason about. There might be some performance hit, but that is really negligible in modern systems.
2. Yes, that still needs to be done on java side. ArrayList will become json array.
3. I explained this one earlier but essentially having a json api allows you to iterate on frontend easily without going to server making changes deploying again etc.